include/m_lights.h

```diff
@@ -4,42 +4,90 @@
 #include "ultra64.h"
 #include "z64math.h"
 #include "unk.h"
+#include "gbi.h"
 
 struct GraphicsContext;
 struct PlayState;
 
+typedef struct LightPoint {
+    /* 0x0 */ s16 x;
+    /* 0x2 */ s16 y;
+    /* 0x4 */ s16 z;
+    /* 0x6 */ u8 color[3];
+    /* 0x9 */ u8 drawGlow;
+    /* 0xA */ s16 radius;
+} LightPoint; // size = 0xC
+
+typedef struct LightDiffuse {
+    /* 0x0 */ s8 x;
+    /* 0x1 */ s8 y;
+    /* 0x2 */ s8 z;
+    /* 0x3 */ u8 color[3];
+} LightDiffuse; // size = 0x6
+
+typedef union {
+    LightPoint point;
+    LightDiffuse diffuse;
+} LightParams; // size = 0xC
+
 typedef struct Lights {
-    /* 0x00 */ UNK_TYPE1 unk_00[0x80];
-} Lights; // size = 0x80
+    /* 0x00 */ u8 type;
+    /* 0x04 */ LightParams lights;
+} Lights; // size = 0x10
+
+typedef struct LightNode {
+    /* 0x0 */ Lights* info;
+    /* 0x4 */ struct LightNode* prev;
+    /* 0x8 */ struct LightNode* next;
+} LightNode; // size = 0xC
+
+typedef struct LightBuffer {
+    /* 0x00 */ s32 current;
+    /* 0x04 */ s32 idx;
+    /* 0x08 */ LightNode lights[32];
+} LightBuffer; // size = 0x188
+
+typedef struct LightsN {
+    /* 0x00 */ u8 diffuse_count;
+    /* 0x08 */ Lightsn lights;
+}LightsN; // size = 0x80
 
 typedef struct LightContext {
-    /* 0x0 */ UNK_TYPE4 unk_0;
-} LightContext; // size >= 0x4
-
-// void func_8009B1E0_jp();
-// void func_8009B23C_jp();
-// void func_8009B2A0_jp();
-// void func_8009B308_jp();
-// void func_8009B33C_jp();
-// void func_8009B394_jp();
-void LightsN_disp(Lights* light, struct GraphicsContext* gfxCtx);
-// void func_8009B504_jp();
-// void func_8009B534_jp();
-// void func_8009B72C_jp();
-// void func_8009B81C_jp();
-void LightsN_list_check(Lights* light, UNK_TYPE4 arg1, Vec3f* arg2);
-// void func_8009B954_jp();
-// void func_8009B9E0_jp();
-// void Global_light_ct();
-// void func_8009BA94_jp();
-// void func_8009BAC0_jp();
-Lights* Global_light_read(LightContext* lightCtx, struct GraphicsContext* gfxCtx);
-// void func_8009BB34_jp();
-// void func_8009BB40_jp();
-// void func_8009BB8C_jp();
-// void func_8009BBEC_jp();
-// void func_8009BC48_jp();
-// void func_8009BD90_jp();
+    /* 0x0 */ LightNode* list;
+    /* 0x4 */ u8 ambientColor[3];
+    /* 0x7 */ u8 fogColor[3];
+    /* 0xA */ s16 fogNear;
+    /* 0xC */ s16 fogFar; 
+} LightContext; // size = 0x10
```

Comment:
Looking at the gcn version this struct should be renamed to `Global_light` instead